### ISSUE TITLE
Code intel and blame errors in repo with spaces with name

### ISSUE THREAD

**Issue by christinelovett** [sourcegraph/sourcegraph#15402]
Sourcegraph 3.21.2
Kubernetes 

Users are facing a series of issues on their Sourcegraph instance (Code is all C#)

- Wonky highlighting (appears to be shifted to the left when hovering) 
- Missing code intelligence and find references fails with errors in the bottom panel 
- Repository status is 'not found' for those with spaces in the name, was supposed to be fixed here: https://github.com/sourcegraph/sourcegraph/pull/14691
- Some symbols don't have hover tooltips but are searchable

We're currently waiting on the frontend and repo-updater logs from the user. 

https://app.hubspot.com/contacts/2762526/company/1085513342

**Comment by Joelkw** [comment]
@macraig I'm adding this to the web team backlog column just so I don't lose track of it, but if this turns out to be a code intel issue (or a split issue) and code intel can handle it, that is much appreciated! (There is another p0 bug the web team is dealing with that just came up as well re: the bitbucket integration for a customer).  In that case, feel free to untag this as web.

**Comment by tjkandala** [comment]
I believe the merged PR for repositories with spaces in the name is https://github.com/sourcegraph/sourcegraph/pull/14681 (https://github.com/sourcegraph/sourcegraph/pull/14691 was an exploratory draft)

**Comment by felixfbecker** [comment]
There definitely seem to be multiple issues going on here. Please make sure to also get us any error messages in the browser console, not just backend logs.

> Wonky highlighting (appears to be shifted to the left when hovering)

This could be https://github.com/sourcegraph/sourcegraph/issues/107. A sample file to reproduce on would be good.

> Missing code intelligence and find references fails with errors in the bottom panel

Could be a web issue or a code intel issue. Can we please get the error messages that are shown?

> Repository status is 'not found' for those with spaces in the name, was supposed to be fixed here:

Could we get the repository name that is failing? Given we added integration tests on this for the web app this may be a backend issue too, may have to ask for screenshots of the network tab of the `RepositoryRedirect` request (the request body and the response). Otherwise it would be good if we had a dogfood instance of their code host to reproduce with. @sourcegraph/distribution do we have that?

> Some symbols don't have hover tooltips but are searchable

This could also be multiple reasons, the best way to know here too would be to also get the request and response body from the `CodeIntelSearch` request in the network tab.

-----

@sourcegraph/web this really makes me wish we had https://github.com/sourcegraph/sourcegraph/issues/9944 to just ask this customer for some browser logs to narrow down the issue.

**Comment by macraig** [comment]
@Joelkw noted! We are waiting to get the logs for further context. cc @efritz @gbrik

**Comment by macraig** [comment]
@christinelovett any updates on the error messages and logs?

**Comment by christinelovett** [comment]
Yes! I heard back last night: 

Still waiting on logs access.

**Re Highlighting:**

I search:
![image](https://user-images.githubusercontent.com/12041606/98560379-b64dc480-2275-11eb-97fd-907c20710e3a.png)

Then click on first line:
![unnamed-1](https://user-images.githubusercontent.com/12041606/98560495-cfef0c00-2275-11eb-899b-f4a37a9e2b22.png)

**Re Errors:**
![image (1)](https://user-images.githubusercontent.com/12041606/98560529-da110a80-2275-11eb-838f-14d70f911475.png)

Upon clicking on the identifier "SubstrateRoutingException" in the source code, an additional error appears:

![unnamed-2](https://user-images.githubusercontent.com/12041606/98560606-edbc7100-2275-11eb-80a2-ad0efe7cc90e.png)


Clicking "Find References" produces:
![image (2)](https://user-images.githubusercontent.com/12041606/98560629-f4e37f00-2275-11eb-985d-01187e5f05bd.png)

and again:
![unnamed-3](https://user-images.githubusercontent.com/12041606/98560662-02006e00-2276-11eb-9c84-86ac85a4f787.png)

**Re RepositoryRedirect:**

Request:


```
{data: {repositoryRedirect: {__typename: "Repository", id: "UmVwb3NpdG9yeTo4",…}}}
data: {repositoryRedirect: {__typename: "Repository", id: "UmVwb3NpdG9yeTo4",…}}
repositoryRedirect: {__typename: "Repository", id: "UmVwb3NpdG9yeTo4",…}
defaultBranch: {displayName: "master"}
description: ""
externalURLs: []
id: "UmVwb3NpdG9yeTo4"
name: "vs-ssh.visualstudio.com/v3/outlookweb/Outlook Web/server-ows-api"
url: "/vs-ssh.visualstudio.com/v3/outlookweb/Outlook Web/server-ows-api"
viewerCanAdminister: true
__typename: "Repository"
```


Response:

```
{"data":{"repositoryRedirect":{"__typename":"Repository",
"id":"UmVwb3NpdG9yeTo4",
"name":"vs-ssh.visualstudio.com/v3/outlookweb/Outlook Web/server-ows-api",
"url":"/vs-ssh.visualstudio.com/v3/outlookweb/Outlook Web/server-ows-api",
"externalURLs":[],
"description":"",
"viewerCanAdminister":true,
"defaultBranch":{"displayName":"master"}}}}
```

Network tab:
![image (3)](https://user-images.githubusercontent.com/12041606/98560861-35db9380-2276-11eb-899e-3b351d8156da.png)


**Re CodeIntelSearch:**

I don't see CodeIntelSearch (IDictionary does not present a tooltip when clicked on):
![image (4)](https://user-images.githubusercontent.com/12041606/98560907-44c24600-2276-11eb-82e4-d5630babd3f7.png)
![image (5)](https://user-images.githubusercontent.com/12041606/98560912-4724a000-2276-11eb-85ec-e13c7e76c646.png)

**Comment by felixfbecker** [comment]
Ugh, looks like there are issues with spaces in paths deep in our extensions URI handling.
@sourcegraph/code-intel do you have unit tests in the extensions for files with spaces?

**Comment by daxmc99** [comment]
> We may have to ask for screenshots of the network tab of the RepositoryRedirect request (the request body and the response). Otherwise it would be good if we had a dogfood instance of their code host to reproduce with. @sourcegraph/distribution do we have that?

We do have the `https://k8s.sgdev.org/search` deployment where we could add a relevant code host. 
The HubSpot link doesn't provide much info though. 

What is their code host?

**Comment by felixfbecker** [comment]
Looking at the repo name above: `vs-ssh.visualstudio.com`
It seems to be Azure DevOps. We could probably just create an org on visualstudio.com and add it as an external service.

**Comment by felixfbecker** [comment]
Created a demo repo here: https://sourcegraph.visualstudio.com/sourcegraph/_git/Test%20Repo

Looking at the docs, it looks like we don't support Azure DevOps?  https://docs.sourcegraph.com/admin/external_service
I assume they use "Other"?

**Comment by felixfbecker** [comment]
Got the test repo working on k8s: https://k8s.sgdev.org/sourcegraph.visualstudio.com/Test%20Repo/-/blob/Test%20Folder/Test%20File.cs

**Comment by felixfbecker** [comment]
Here's the fix for the git blame error: https://github.com/sourcegraph/sourcegraph-git-extras/pull/155
It will be available to them right after it's merged & published without having to upgrade.

**Comment by felixfbecker** [comment]
Narrowed the codeintel down to being inside the code intel extension: https://github.com/sourcegraph/code-intel-extensions/issues/526 @efritz @macraig

**Comment by felixfbecker** [comment]
I also discovered https://github.com/sourcegraph/sourcegraph/issues/15615 which they didn't report but will affect them. cc @Joelkw @jeanduplessis

**Comment by felixfbecker** [comment]
The references panel errors are a web team issue, filed: https://github.com/sourcegraph/sourcegraph/issues/15618

**Comment by christinelovett** [comment]
Thanks @felixfbecker - and yes they're using Azure DevOps as Other

**Comment by efritz** [comment]
https://github.com/sourcegraph/code-intel-extensions/pull/528 should fix these issues in the code intel extension. I think everything else is owned by web (please let me know if I missed something).

**Comment by felixfbecker** [comment]
I narrowed down the other issue I found to an issue in the backend API: https://github.com/sourcegraph/sourcegraph/issues/15667
Assigned to @sourcegraph/cloud

**Comment by felixfbecker** [comment]
References bug is fixed and will go out in the next release (3.22).

**Comment by felixfbecker** [comment]
All linked issues are fixed, so considering this fixed.
